### PR TITLE
Feature/improved documentation

### PR DESCRIPTION
**Pull request by FrankTN**
I started adding documentation to the UI dialog classes. I will continue working on the UI documentation.

### CODE REVIEW

**Comment by FrankTN** [comment]
Interesting, the build worked locally.

**Comment by FrankTN** [comment]
Verified again that the build and tests run locally, restarting in Travis

File: Hestia/app/src/main/java/hestia/backend/NetworkHandler.java [hunk 1]
```diff
@@ -33,7 +34,7 @@
  * as well as getters and setters for the ip and the port number.
  */
 
-public class NetworkHandler extends Application {
+public class NetworkHandler extends Application implements Serializable{
```
Comment (StefanEvanghelides) [inline Hestia/app/src/main/java/hestia/backend/NetworkHandler.java:37]: why is this needed?

File: Hestia/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -49,4 +49,7 @@
     <string name="passSet">Password set to: </string>
     <string name="userSet">Username set to: </string>
     <string name="userNotSet">Username not changed (length under 5)</string>
+    <string name="cancelNameChange">Name of the device was not changed</string>
+    <string name="titleNameChange">Change name</string>
+    <string name="cancelIPChange">\"Cancel pressed\"</string>
```
Comment (StefanEvanghelides) [inline Hestia/app/src/main/res/values/strings.xml:54]: "Cancel pressed" is quite generic, so I believe we should not give it a specific name as "cancelIPChange". I think that "cancelPressed" would do. Or maybe change the string from "Cancel Pressed" to something more related to the IP.

**Comment by StefanEvanghelides** [comment]
Good changes. It is a pitty that those tests fail Travis even though they run locally
